x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -62,7 +62,16 @@ export async function getServiceAnomalies(
     return [];
   }
 
-  const { jobs: apmMlJobs } = await ml.anomalyDetectors.jobs('apm');
+  let apmMlJobs: AnomalyDetectionJob[] = [];
+  try {
+    const { jobs } = await ml.anomalyDetectors.jobs(APM_ML_JOB_GROUP_NAME);
+    apmMlJobs = jobs;
+  } catch (error) {
+    // did not find any apm jobs
```

Comment:
Thinking more about this: I don't think we should silently swallow any error that occurs. Would be better if we can catch the error and only return under the right circumstances, and re-throw if not.

Reply:
WDYT about something like:

```ts
function getAnomalyDetectionJobs(): AnomalyDetectionJob[] {
  try {
    const { jobs } = await ml.anomalyDetectors.jobs(APM_ML_JOB_GROUP_NAME);
    return
  } catch (error) {
    if(error.code === 404) {
      return []
    }
    throw error
  }
}
```

Reply:
> I don't think we should silently swallow any error that occurs. Would be better if we can catch the error and only return under the right circumstances, and re-throw if not.

+10000 to this.